### PR TITLE
Test framework Student View - Displaying Test Results

### PR DESCRIPTION
**Pull request by jeremygoh**
First pass at cleaning up the student view for displaying test results in JSON format in reverse chronological order.

This involved
- adding created_at and update_at timestamps to the test_results table
- correcting the method for calculating marks earned
- formatting the date and adding appropriate translations
- styling the view (it was missing any CSS styling before)

Before
screen shot 2015-11-29 at 11 26 18 pm

After
screen shot 2015-11-29 at 7 42 42 pm

### CODE REVIEW

File: db/migrate/20151129233225_add_timestamps_to_test_result.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AddTimestampsToTestResult < ActiveRecord::Migration
+  def change
+    change_table :test_results do |t|
```
Comment (jeremygoh) [inline db/migrate/20151129233225_add_timestamps_to_test_result.rb]: I'm not sure I buy this suggestion...

Comment (david-yz-liu) [inline db/migrate/20151129233225_add_timestamps_to_test_result.rb]: Okay that's fine.

File: app/views/automated_tests/student_interface.html.erb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 <% # UI LIBRARIES %>
 <%= javascript_include_tag "livepipe/livepipe.js",
   "livepipe/window.js" %>
+<%= stylesheet_link_tag '//cdnjs.cloudflare.com/'\
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: You can remove these. I don't mind if it isn't pretty.

File: app/views/automated_tests/student_interface.html.erb [hunk 3]
```diff
@@ -71,62 +76,24 @@
...
-			  	    <% end %>
-		  	    </ul>
-		  	    <ul> </ul>
-		  	  <% end %>
-		  	  </ul>
-		  	</li>
-		  <% end %>
-		  </ul>
-		</div>
+    <div class="block_content">
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: It seems like the indentation for this is off.

File: app/views/automated_tests/student_interface.html.erb [hunk 4]
```diff
@@ -71,62 +76,24 @@
...
+  	  <br />
+  	  <div class="sub_block">
+  	  	<%= raw ( I18n.t("automated_tests.marks_obtained",
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: Don't use `I18n.t`, you can just use `t`.

File: app/views/automated_tests/student_interface.html.erb [hunk 5]
```diff
@@ -71,62 +76,24 @@
...
+  	   	          	      :marking => "#{@grouping.get_total_test_script_marks * 1.0} / #{@assignment.total_ror_script_marks * 1.0}" ) ) %>
+  	  </div>
+  	  <div id="test_results" class="sub_block">
+        <% @test_script_results.each do |test_script_result| %>
+          <h3><%= I18n.t("automated_tests.test_run") %> -
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: Make this into a separate partial.

File: app/views/automated_tests/student_interface.html.erb [hunk 6]
```diff
@@ -71,62 +76,24 @@
...
+            <%= l(test_script_result.created_at, format: :long)%></h3>
+          <pre>
+            <code class="js">
+              <%= "\n#{JSON.pretty_generate(JSON.parse(test_script_result.actual_output))}\n" %>
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: It's fine to just leave it as `test_script_result.actual_output` for now.

File: app/views/automated_tests/student_interface.html.erb [hunk 7]
```diff
@@ -71,62 +76,24 @@
...
+            </code>
+          </pre>
+        <% end %>
+  		</div>
```
Comment (david-yz-liu) [inline app/views/automated_tests/student_interface.html.erb]: This div is in a weird spot.

File: app/helpers/automated_tests_helper.rb [hunk 8]
```diff
@@ -5,6 +5,14 @@ module AutomatedTestsHelper
   # it is enqueued and it is waiting for execution. Resque manages this queue.
   @queue = :test_waiting_list
 
+  def fetch_latest_tokens_for_grouping(grouping)
```
Comment (jeremygoh) [inline app/helpers/automated_tests_helper.rb]: Resolved. Hound just didn't pick this one up.

**Comment by jeremygoh** [comment]
@david-yz-liu I've made the changes requested as well as split up the route for viewing the student interface from running the tests.

The rational behind this was to enable users to be redirected to the student_interface url after running a test and allow them to refresh the page without starting a new test run.

**Pull request merged**